stdlib/2/os/path.pyi

```diff
@@ -22,25 +28,31 @@ devnull = ...  # type: str
 def abspath(path: AnyStr) -> AnyStr: ...
 def basename(path: AnyStr) -> AnyStr: ...
 
-def commonprefix(list: List[AnyStr]) -> AnyStr: ...
+if sys.version_info >= (3, 5):
+    def commonpath(paths: Sequence[AnyStr]) -> AnyStr: ...
+if sys.version_info >= (3,):
+    # NOTE: Empty List[bytes] results in '' (str) => fall back to Any return type.
+    def commonprefix(list: List[AnyStr]) -> Any: ...
+else:
+    def commonprefix(list: List[AnyStr]) -> AnyStr: ...
```

Comment:
Do you mean `List[Union[bytes, Text]]` in the second overload?

I think we can just do `def commonprefix(list: Sequence[Union[bytes, Text]]) -> Any: ...` (not a List so we don't get hit by invariance).